src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -29,20 +28,20 @@ namespace MongoDB.Driver.Core.Servers
 {
     internal sealed class ServerMonitor : IServerMonitor
     {
-        private static readonly TimeSpan __minHeartbeatInterval = TimeSpan.FromMilliseconds(500);
-
-        private readonly ExponentiallyWeightedMovingAverage _averageRoundTripTimeCalculator = new ExponentiallyWeightedMovingAverage(0.2);
         private readonly ServerDescription _baseDescription;
-        private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
         private volatile IConnection _connection;
         private readonly IConnectionFactory _connectionFactory;
+        private CancellationTokenSource _currentCheckCancellationTokenSource;
```

Comment:
Consider renaming to (and adding a comment):

    private CancellationTokenSource _heartbeatCancellationTokenSource; // used to cancel an ongoing heartbeat

Reply:
done